packit/utils/lookaside.py

```diff
@@ -0,0 +1,47 @@
+import configparser
+
+from pathlib import Path
+from typing import Dict, List, Union
+
+import pyrpkg
+
+from packit.exceptions import PackitLookasideCacheException
+
+
+def get_lookaside_sources(
+    pkg_tool: str,
+    package: str,
+    basepath: Union[Path, str],
+) -> List[Dict[str, str]]:
+    """
+    Gets URLs to sources stored in lookaside cache.
+
+    Args:
+        pkg_tool: Packaging tool associated with a lookaside cache instance
+          to be used to get the URLs.
+        package: Package name.
+        basepath: Path to a dist-git repo containing the "sources" file.
+
+    Returns:
+        List of dicts with path (filename) and URL.
+    """
+    try:
+        parser = configparser.ConfigParser()
+        parser.read(f"/etc/rpkg/{pkg_tool}.conf")
+        config = dict(parser.items(pkg_tool, raw=True))
+        cache = pyrpkg.lookaside.CGILookasideCache(
+            config["lookasidehash"], config["lookaside"], config["lookaside_cgi"]
+        )
+    except (configparser.Error, KeyError) as e:
+        raise PackitLookasideCacheException("Failed to read rpkg configuration") from e
+    try:
+        sources = pyrpkg.sources.SourcesFile(Path(basepath) / "sources", "bsd")
+    except (pyrpkg.errors.MalformedLineError, ValueError) as e:
+        raise PackitLookasideCacheException("Failed to parse sources") from e
+    result = []
+    for entry in sources.entries:
+        url = cache.get_download_url(
+            f"rpms/{package}", entry.file, entry.hash, entry.hashtype
+        )
+        result.append({"path": entry.file, "url": url})
+    return result
```

Comment:
I wish all of this code was part of rpkg - lots of sorcery here.

Reply:
True, especially parsing the config files.